### PR TITLE
adding in a smart mastering search tab

### PR DESCRIPTION
**Pull request by paxtonhare**

### CODE REVIEW

File: quick-start/src/main/java/com/marklogic/quickstart/service/SmartMasteringService.java [hunk 1]
```diff
@@ -5,26 +5,42 @@
...
 import com.marklogic.client.util.RequestParameters;
-import com.marklogic.hub.DatabaseKind;
 import com.sun.jersey.api.client.ClientHandlerException;
 import org.springframework.stereotype.Service;
 
 @Service
 public class SmartMasteringService {
 
     private final String MASTERING_STATS = "mastering-stats";
+    private final String MASTERING_MERGE = "am-merge";
```
Comment (dmcassel) [inline quick-start/src/main/java/com/marklogic/quickstart/service/SmartMasteringService.java]: service name has changed to "sm-merge" as of 0.0.4

File: quick-start/src/main/ui/app/pipes/pipes.module.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+@NgModule({
+  declarations: [
+    ObjectToArrayPipe,
+    TruncateCharactersPipe
+  ],
+  exports: [
+    ObjectToArrayPipe,
+    TruncateCharactersPipe
+  ]
+  // providers: [ObjectToArrayPipe]
```
Comment (dmcassel) [inline quick-start/src/main/ui/app/pipes/pipes.module.ts]: dead code

File: quick-start/src/main/ui/app/smart-mastering/search/search.component.scss [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+.result {
+  padding: 20px;
+  margin: 5px;
+  border: 1px rgba(0,0,0,0.15) solid;
+  width: 300px;
+  box-shadow: 0 1px 3px 0 rgba(0,0,0,.2),0 1px 1px 0 rgba(0,0,0,.14),0 2px 1px -1px rgba(0,0,0,.12);
+  position: relative;
+  // border-bottom: 1px solid rgba(128,128,128,0.15);
```
Comment (dmcassel) [inline quick-start/src/main/ui/app/smart-mastering/search/search.component.scss]: dead code

File: quick-start/src/main/ui/app/smart-mastering/search/search.component.scss [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+    a {
+      color: $link-color;
+      cursor: pointer;
+    }
+  }
+}
+
+.database-chooser {
+  /deep/ mdl-checkbox {
```
Comment (dmcassel) [inline quick-start/src/main/ui/app/smart-mastering/search/search.component.scss:107]: not familiar with "/" as part of a selector. on purpose?

**Review by dmcassel — CHANGES_REQUESTED** [review]

File: quick-start/src/main/ui/app/smart-mastering/search/search.component.scss (hunk 4 above)
Comment (paxtonhare) [reply quick-start/src/main/ui/app/smart-mastering/search/search.component.scss:107]: yep. it's an angular thing

**Review by dmcassel — APPROVED** [review]

**Pull request merged**